### PR TITLE
crimson: cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/osd/object_context_list.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (ronen-fr) [inline src/crimson/osd/object_context_list.h:1]: how is the change related? (no criticism - just trying to understand...)

Comment (tchaikov) [reply src/crimson/osd/object_context_list.h:1]: thanks for pointing this out! this change should not be included in this PR.

File: src/test/crimson/seastore/test_seastore.cc [hunk 2]
```diff
@@ -115,15 +115,15 @@ struct seastore_test_t :
 
 	if (!iter->valid() || refiter->first < iter->key()) {
 	  logger().debug(
-	    "check_omap: misisng omap key {}",
+	    "check_omap: missing omap key {}",
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/test/crimson/seastore/test_seastore.cc:118]: @ronen-fr could you be more specific? which PR are you referencing?

[the remark this answers is not in the document]
Comment (tchaikov) [reply src/test/crimson/seastore/test_seastore.cc:118]: ahh, I see. I managed to include this change in #39992.

drop from it. it was a mistake.

thank you!

File: src/crimson/os/seastore/onode.h [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 namespace crimson::os::seastore {
 
-struct __attribute__((packed)) onode_layout_t {
+struct onode_layout_t {
   ceph_le32 size{0};
   omap_root_le_t omap_root;
 } __attribute__((packed));
```
Comment (rzarzynski) [inline src/crimson/os/seastore/onode.h:19]: ACK.

**Review by rzarzynski — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- dropped the cmake changes. it turns out the seastore's dependencies between different subcomponents are more involved than i expected.

**Pull request merged**